src/test/scala/nl.knaw.dans.easy.ccw/CollectSpec.scala

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (C) 2017 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.ccw
+
+import better.files.File
+import nl.knaw.dans.easy.ccw.Command.configuration
+import org.apache.commons.configuration.PropertiesConfiguration
+import org.scalatest.BeforeAndAfterAll
+
+import scala.language.postfixOps
+import scala.util.Success
+
+class CollectSpec extends TestSupportFixture with BeforeAndAfterAll {
+
+  val resourceDir = File(getClass.getResource("/"))
+  val configuration = new Configuration("version x.y.z",
+    new PropertiesConfiguration() {},
+    new PropertiesConfiguration() {
+      setDelimiterParsingDisabled(true)
+      load(resourceDir / "debug-config" / "datamanager.properties" toJava)
+    })
+  val datamanagerProperties = configuration.datamanagers
+  val commonCurationArea = testDir / "easy-common-curation-area"
+  val datamanagerCurationAreas = testDir / "datamanager-curation-areas"
+  val managerCurationDir = datamanagerCurationAreas / "$unix-user/curation-area" toString
+  def managerCurationArea(datamanager: DatamanagerId) = File(managerCurationDir.replace("$unix-user", datamanager))
+  val jannekesCurationArea = datamanagerCurationAreas / "janneke/curation-area"
+  val testLog = testDir / ".." / "test.log"
+
+  val collector = new EasyCollectCurationWorkApp(commonCurationArea, managerCurationArea, datamanagerProperties)
+
+  override def beforeAll(): Unit = {
+    testLog delete (true)
+  }
+
+  override def beforeEach(): Unit = {
+    commonCurationArea delete (true)
+    jannekesCurationArea delete (true)
+    File(getClass.getResource("/easy-common-curation-area")) copyTo commonCurationArea
+    File(getClass.getResource("/datamanager-curation-areas/janneke/curation-area")) copyTo jannekesCurationArea
+    commonCurationArea.toJava should exist
+    jannekesCurationArea.toJava should exist
+  }
+
+  "collect" should "succeed" in {
+    collector.run() shouldBe a[Success[_]]
```

Comment:
you got an interesting test style: run the actual command in the first test and test all kinds of side effects in successive tests, without running the actual command again. All the while you delete and create stuff for each test.

The major concern here is that tests will fail now when they run in a different order. A test should be self-contained and not depend on the results/side-effects of other tests.

I would suggest the following:
* move the `testLog delete (true)` to the `beforeEach` (with that you don't need to override `beforeAll`, neither do you need the corresponding mixin trait)
* run the `collector.run()` command for every test that you need it to and perform the tests you want to do afterwards

Btw, I like the tests you're doing. They're good, so no need to change them.

Reply:
done